init.lua

```diff
@@ -19,15 +19,38 @@
 -- Thank you, ApolloX, for adding this.
 -- ApollloX: No problem, glade to help.
 
-local player_lights = {}
+goodtorch = {} -- Public API
+goodtorch.modpath = minetest.get_mod_path("goodtorch")
+
+goodtorch.player_lights = {}
+goodtorch.nodes = { -- Replacement nodes (excluding our own nodes, as they don't change)
+	-- Don't need air as air is always air. (That's a lot of air)
+	water = {
+		source = "",
+		flowing = "" -- Uses param2
+	},
+	river = {
+		source = "",
+		flowing = "" -- Uses param2
+	}
+}
```

Comment:
Common occurence in mods, a public api (which for our case exposes nodes, player_lights and modpath, which really don't need to be exposed but have to for cross file coding)

Our "api" consists of 3 functions:

- detect_gamemode()
- default_nodeset()
- mcl_nodeset()

Mostly used to bring some kind of fancy interchangable code based on what "game" mode we are running in.